### DISCUSSION TITLE
Equivalent to #| include: false for markdown cells?

### DISCUSSION THREAD

**Discussion by jlperla** [quarto-dev/quarto-cli#6906] [Q&A] [answered]
### Description

I have a notebook where I want to manage statements of problem set quetsions as well as the solutions, but easily flip off the solutions before giving the assignment.  For code, I am doing something like the following:
    
    ```{python}
    # tell people to modify answer here
    ```
    
    ```{python}
    #| include: false
    # add my solution
    ```

And then I switch the `false` to `true` for my answer key when generating the jupyter notebooks

For more text based quetsions I am adding in something like the following, which tells people to doubleclick on a cell to add an answer like the following:
    
      
    ::: {.cell .markdown}
    
    (double click to edit your answer)
    
    :::

The problem is that I don't know how to add in a `::: {.cell .markdown}` for the solution which I can easily flip on/off like with the `#| include: false`.

Any tricks here?

**Comment by cscheid** [comment, marked as answer]
Try this: https://quarto.org/docs/authoring/conditional.html

**Comment by jlperla** [comment]
Perfect, thanks.  For anyone else who reads this, here is my full solution:

    ## Question
   Ask some question to be edited in the jupyter notebook
  
    :: {.cell .markdown}
    
    (double click to edit your answer)
    
    :::
    
    ::: {.cell .markdown .content-visible}
    
    ### MY SOLUTION
    ... some answer
    
    :::

Then I use a bash script to swap content-visible to content-hidden and back when calling the notebook rendering

**Reply by cderv** [reply]
You could have a look at [project profile](https://quarto.org/docs/projects/profiles.html#profile-content)

This would allow you to pass a `--profile` flag for your rendering, and have some content act accordingly. Could prevent usage of bash script and file modification.

**Comment by jlperla** [comment]
Very cool.  How do you use the `profile` thing in a code block.  e.g. I am currently doing

    ```{python}
    #| include: false
    # problem set solution
    x = 3 # etc.
    ```

And flipping that `#| include`

Edit: or can you wrap that whole code block in a cell?  I mostly care about jupyter output here

**Reply by cderv** [reply]
I don't think you can map a profile to `include` directly, and we may be missing a good way to do that. 

> Edit: or can you wrap that whole code block in a cell? I mostly care about jupyter output here

This should work, but with one slight side effect. Chunk is executed, but code and result not inserted in the output. Which you may want. 

Accessing profile info in code can be done with `QUARTO_PROFILE` env var.
